xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -2455,12 +2455,14 @@ void CVideoPlayer::OnExit()
   CFFmpegLog::ClearLogLevel();
   m_bStop = true;
 
-  if (m_error)
-    m_callback.OnPlayBackError();
-  else if (m_bAbortRequest)
-    m_callback.OnPlayBackStopped();
-  else
-    m_callback.OnPlayBackEnded();
+  CJobManager::GetInstance().Submit([this]() {
```

Comment:
Is `this` guaranteed to exist until the job is complete? Or should we only capture the callback, given that the callback is CApplication which has an appropriate lifetime?

Reply:
good question. I think we should use copes and no refs passing to the lambda

Reply:
updated